scripts/generate_ics.py

```
@@ -0,0 +1,183 @@
+        # Generate output filename
+        year_part = args.year.replace("-", "_") if args.year else "full"
+        file_name = args.output or f"{args.code.upper()}_{year_part}.ics"
+
+        if os.path.exists(file_name):
+            logger.warning("Overwriting existing file: %s", file_name)
+
+        exporter.save_ics(file_name)
+
+        logger.info("Calendar exported to: %s", file_name)
+
+    except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)